dali/pipeline/operator/checkpointing/checkpoint.cc

```diff
@@ -41,4 +43,40 @@ size_t Checkpoint::GetIterationId() const {
   return iteration_id_;
 }
 
+Index Checkpoint::NumOp() const {
+  return cpts_.size();
+}
+
+std::string Checkpoint::SerializeToProtobuf(const OpGraph &graph) const {
+  dali_proto::Checkpoint checkpoint;
+  const auto &nodes = graph.GetOpNodes();
+  for (size_t i = 0; i < nodes.size(); i++) {
+    auto op_cpt = checkpoint.add_cpts();
+    op_cpt->set_operator_name(cpts_[i].OperatorName());
+    op_cpt->set_operator_state(nodes[i].op->SerializeCheckpoint(cpts_[i]));
+  }
+  return checkpoint.SerializeAsString();
+}
+
+void Checkpoint::DeserializeFromProtobuf(const OpGraph &graph,
+                                         const std::string &serialized_data) {
+  Build(graph);
+  dali_proto::Checkpoint checkpoint;
+  checkpoint.ParseFromString(serialized_data);
+  DALI_ENFORCE(checkpoint.cpts_size() == static_cast<int>(cpts_.size()),
+               "Provided checkpoint stores state of different number of operators "
+               "than there are in the pipeline. ");
```

Comment:
A bit of a nitpick, but wouldn't it be clearer this way?

```suggestion
               "The number of operators in the checkpoint differs from the number of operators in the pipeline.");
```